### PR TITLE
Fixes #15747: When receving pleinty of inventories at the same time, the web interface starts to parse them all at once (branch 5.0)

### PR DESCRIPTION
**Pull request by ncharles**
https://issues.rudder.io/issues/15747

### CODE REVIEW

**Comment by ncharles** [comment]
Commit modified

File: webapp/sources/ldap-inventory/inventory-provisioning-web/src/main/scala/com/normation/inventory/provisioning/endpoint/InventoryFileWatcher.scala [hunk 1]
```diff
@@ -234,8 +234,10 @@ class ProcessFile(
   private def remove(file: File): Unit = synchronized {
     toBeProcessed = (toBeProcessed - file)
   }
-
-  val newTask = (f:File) => Task (processFile(f) ).delayExecution(500 millis).doOnFinish (_ => Task (remove(f)))
+  // We cannot handle too many inventories at the same time, so we:
+  // * increase by 500ms the delay per inventories in the list of inventories to be read
+  // * increase by 3s the delay per inventory in the queue of the Inventory Processor
+  val newTask = (f:File) => Task (processFile(f) ).delayExecution((toBeProcessed.size + 1) * 500 + (inventoryProcessor.queueSize.get) * 3000 millis).doOnFinish (_ => Task (remove(f)))
```
[the remark this answers is not in the document]
Comment (fanf) [reply webapp/sources/ldap-inventory/inventory-provisioning-web/src/main/scala/com/normation/inventory/provisioning/endpoint/InventoryFileWatcher.scala]: I don't see any better way. I propose to merge that, and see how it behaves in tests.

**Comment by fanf** [comment]
OK, merging this PR

File: webapp/sources/ldap-inventory/inventory-provisioning-web/src/main/scala/com/normation/inventory/provisioning/endpoint/InventoryFileWatcher.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (ncharles) [reply webapp/sources/ldap-inventory/inventory-provisioning-web/src/main/scala/com/normation/inventory/provisioning/endpoint/InventoryFileWatcher.scala]: there's no easy solution (appart from changing everything like you did in 5.1) - it improves the situation, but is in no a total fix. It has the benefit of smoothing out incompletly the inventory parsing

**Pull request merged**